### PR TITLE
lock user by admin

### PR DESCRIPTION
**Pull request by nguyenvanson332000**
## Related Tickets
- [#44389](https://edu-redmine.sun-asterisk.vn/issues/44389)

## WHAT (optional)
- Lock and search user by admin

## HOW
- lock and unlock user by admin with gem devise
- search role, name and email user by admin with gem ransack 

## Evidence (Screenshot or Video)
ảnh
ảnh

### CODE REVIEW

**Comment by nguyenvanson332000** [comment]
ready

File: app/javascript/packs/handle_banner.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      type: 'PATCH',
+      url: "/admin/users/"+$(this).data().id,
+      beforeSend: function(xhr) {xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'))},
+      data: {
+        id: $(this).data().id,
+        type: $(this).data().type
+      },
+      dataType: 'JSON'
+    }).done(function (data) {
+      alert("Xu ly thanh cong");
```
Comment (ghost) [inline app/javascript/packs/handle_banner.js]: i18n em ơi

File: app/views/admin/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        <th scope="col">
+          <%= t "show.role" %>
+        </th>
+        <th scope="col">
+          <%= t "booking.action" %>
+        </th>
+      </tr>
+    </thead>
+    <tbody>
+      <% @users.each do |user| %>
```
Comment (ghost) [inline app/views/admin/users/index.html.erb]: Dùng partial đi em

**Review by ghost — COMMENTED** [review]
cmt

File: app/javascript/packs/handle_banner.js (hunk 1 above)
Comment (nguyenvanson332000) [reply app/javascript/packs/handle_banner.js]: dạ vâng ạ

File: app/views/admin/users/index.html.erb (hunk 2 above)
Comment (nguyenvanson332000) [reply app/views/admin/users/index.html.erb]: dạ

**Comment by nguyenvanson332000** [comment]
ready

File: app/javascript/packs/application.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ require("@rails/activestorage").start()
 require("channels")
 require("jquery")
 import "bootstrap"
-import "application/sweet-alert-confirm"
+// import "application/sweet-alert-confirm"
```
Comment (ghost) [inline app/javascript/packs/application.js]: Không dùng thì xóa dòng này đi em

File: app/views/admin/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<% provide(:title, t("menu_manager.booking_field")) %>
+<% provide(:base_title, t("base.title")) %>
```
Comment (ghost) [inline app/views/admin/users/index.html.erb]: bỏ bớt ()

**Review by ghost — COMMENTED** [review]
cmt

File: app/views/admin/users/index.html.erb (hunk 4 above)
Comment (nguyenvanson332000) [reply app/views/admin/users/index.html.erb]: dạ

File: app/javascript/packs/application.js (hunk 3 above)
Comment (nguyenvanson332000) [reply app/javascript/packs/application.js]: dạ

**Comment by nguyenvanson332000** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**